common/persistence/tests/mysql_test.go

```diff
@@ -617,3 +619,23 @@ func TestMySQLClosedConnectionError(t *testing.T) {
 	s := newConnectionSuite(t, testData.Factory)
 	suite.Run(t, s)
 }
+
+func TestMySQLQueueV2(t *testing.T) {
+	testData, tearDown := setUpMySQLTest(t)
+	t.Cleanup(tearDown)
+
+	queue, err := testData.Factory.NewQueueV2()
+	if err != nil {
+		t.Fatalf("unable to create Queue: %v", err)
+	}
+	t.Run("Generic", func(t *testing.T) {
+		t.Parallel()
+		RunQueueV2TestSuite(t, queue)
+	})
+	t.Run("SQL", func(t *testing.T) {
+		t.Parallel()
+		db, err := testData.Factory.GetDB()
+		require.NoError(t, err)
+		tests.RunSQLQueueV2TestSuite(t, db)
+	})
```

Comment:
This code is identical across all the SQL implementations, right? If so, I think you should extract a method for this which just accepts a test and a factory.

Reply:
Done